packages/react-data-grid/src/utils/viewportUtils.ts

```diff
@@ -60,21 +60,29 @@ export function getHorizontalRangeToRender<R>({
   let { viewportWidth } = columnMetrics;
 
   let remainingScroll = scrollLeft;
-  let columnIndex = lastFrozenColumnIndex;
+  let columnIndex = Math.max(lastFrozenColumnIndex, 0);
   let hiddenColumnsWidth = 0;
   while (remainingScroll >= 0 && columnIndex < columns.length) {
-    columnIndex++;
-    const column = columns[columnIndex];
```

Comment:
@MayhemYDG even with your change above, this will fail because the end validation inside while() does nothing, because the columnIndex is incremented after validating the end condition
`columnIndex < columns.length` : `0 < 1`
`columnIndex++` : `1`
`columns[columnIndex` : `columns[1] // invalid`